### ISSUE TITLE
False positive "must have a body" error

### ISSUE THREAD

**Issue by octylFractal** [intellij-rust/intellij-rust#10804]
## Environment

* **IntelliJ Rust plugin version:** 0.4.200.5421-232
* **Rust toolchain version:** 1.71.0 (8ede3aae2 2023-07-12) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2023.2 (CL-232.8660.186)
* **Operating system:** Linux 6.4.8-zen1-1-zen
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr

## Problem description
Using [`ref-cast`](https://github.com/dtolnay/ref-cast)'s `#[ref_cast_custom]`, a function may not need to have a body as it is implemented by the macro. However, I assume intellij-rust cannot detect this, and emits an error anyways. I would like to be able to turn off this inspection, or have it fixed to detect this.

## Steps to reproduce

1. Add `ref-cast = "1.0"` as a dependency.
2. Add the following file:

```rust
#[derive(RefCastCustom, Debug)]
#[repr(transparent)]
pub struct ClassName {
    raw: str,
}

impl ClassName {
    fn ref_cast(s: &str) -> &Self; // Error here, Associated function `ref_cast` must have a body
}
```

**Comment by Maksim20023** [comment]
Hi! Thank you for the report.
Based on the description, I successfully reproduced this issue in my environment and forwarded your case and steps to reproduce it to our developer's team. Unfortunately, there is no way to suppress this warning and this is another bug wich already known to us. We currently don't have information about ETA for both this issue.
When we have any new information about them, we will add a reply. If you have any further suggestions, please feel free to reach out.
